operator/pkg/ciliumendpointslice/manager.go

```diff
@@ -368,6 +371,110 @@ func (c *cesMgr) RemoveCEPFromCache(cepName string, baseDelay time.Duration) {
 	return
 }
 
+// CompressCESsOnFCFSStartUp makes sure that once FCFS (First Come, First
+// Served) CES manager starts, all CESs will be compressed to a minimum number
+// of fully packed CESs.
+//
+// This is only required when changing the batching strategy from Identity to
+// FCFS, to ensure expected performance. Otherwise, it will perform no action,
+// as all CESs are already compressed by design with FCFS batching.
+func (c *cesMgr) CompressCESsOnFCFSStartUp() {
+	log.Info("Running CompressCESsOnFCFSStartUp.")
+	// The compression needs to be done per namespace, since one CES cannot
+	// contain CEPs with different namespaces.
+	cesPerNamespace := make(map[string][]*cesTracker)
+	for _, ces := range c.desiredCESs.desiredCESs {
+		cesPerNamespace[ces.ces.Namespace] = append(cesPerNamespace[ces.ces.Namespace], ces)
+	}
+
+	for namespace, cesList := range cesPerNamespace {
+		c.compressCESList(cesList, namespace)
+	}
+
+	log.Info("CompressCESsOnFCFSStartUp completed.")
+}
+
+func (c *cesMgr) compressCESList(cesList []*cesTracker, namespace string) {
+	if len(cesList) == 0 {
+		log.WithFields(logrus.Fields{
+			logfields.K8sNamespace: namespace,
+		}).Info("No action required for compressCESList. No CES currently exist.")
+		return
+	}
+
+	totalCEPs := 0
+	for _, ces := range cesList {
+		ces.backendMutex.RLock()
+		totalCEPs += len(ces.ces.Endpoints)
+		ces.backendMutex.RUnlock()
+	}
+
+	// Calculate how many fully packed CES are needed.
+	requiredCESs := int(math.Ceil(float64(totalCEPs) / float64(c.maxCEPsInCES)))
```

Comment:
Are changes to CESes possible when this code run?

Reply:
Yes.
It will not affect the outcome, as when it tries to move CEPs from one CES, it uses the same method as if it was inserted, and it would just create a new CES to add it to.